crates/bevy_core_pipeline/src/tonemapping/mod.rs

```diff
@@ -54,19 +84,98 @@ pub struct TonemappingPipeline {
     texture_bind_group: BindGroupLayout,
 }
 
-#[derive(Copy, Clone, PartialEq, Eq, Hash)]
+#[derive(
+    Component,
+    Debug,
+    Hash,
+    Clone,
+    Copy,
+    Reflect,
+    Default,
+    ExtractComponent,
+    PartialEq,
+    Eq,
+    FromReflect,
+)]
+#[extract_component_filter(With<Camera>)]
+#[reflect(Component)]
+pub enum Tonemapping {
```

Comment:
Feels very odd that this is not user-extensible in any way. It really feels like this should use a trait query instead eventually.

Reply:
Agreed. Though I would think to make this truly extensible we would also need to make it possible for the user to hook in their own shader code to implement other tonemappers correct?